### PR TITLE
Adding type information for TypeScript

### PR DESCRIPTION
**Pull request by Xapphire13**
@maxbrunsfeld, I left a `// TODO` in there for the one thing I couldn't figure out, could you provide insight?

### CODE REVIEW

File: node-tree-sitter.d.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    public getLogger(): LoggerFunc;
+    public setLogger(logFunc: LoggerFunc): void;
+    public getInput(): StringInput | null;
+    public setInput(input: StringInput): void;
+    public edit(delta: EditDelta): Document;
+    public invalidate(): void;
+  }
+
+  // TODO Need to specify the arguments
+  type LoggerFunc = (...args: any[]) => void;
```
Comment (maxbrunsfeld) [inline node-tree-sitter.d.ts]: The logger function takes 3 arguments: a message string, an object containing arbitrary key/value parameters (where the values are strings), and a string that's either `parse` or `lex`, depending on which component emitted the message.

So maybe this:

```ts
type LoggerFunc = (message: string, params: {[param: string]: string}, type: "parse" | "lex") => void;
```

File: package.json [hunk 2]
```diff
@@ -13,6 +13,7 @@
     "lexer"
   ],
   "main": "index.js",
+  "types": "node-tree-sitter.d.ts",
```
Comment (maxbrunsfeld) [inline package.json]: How does this field get used? I'm not very familiar with typescript.

File: node-tree-sitter.d.ts (hunk 1 above)
Comment (Xapphire13) [reply node-tree-sitter.d.ts]: Looks good! I'll update the PR, thanks =]

File: package.json (hunk 2 above)
Comment (Xapphire13) [reply package.json]: The TypeScript compiler will automatically load types defined in the file listed there. It does this when it resolves the `import` statement. [Documentation](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html)

File: node-tree-sitter.d.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    column: number;
+  };
+
+  export type StringInput = {
+    position: number;
+    string: any;
+    bufferSize: number | null;
+    seek(distance: number): void;
+    read(): any;
+  }
```
Comment (maxbrunsfeld) [inline node-tree-sitter.d.ts]: I think that `StringInput` should just be called `Input`, and it should be typed as follows:

```ts
export interface Input = {
  seek(index: number, position: Position): void;
  read(): String;
};
```

The only required methods are `seek` and `read`; all these other properties were just implementation details used in tests or in particular implementations (e.g. `StringInput`).

File: package.json (hunk 2 above)
Comment (maxbrunsfeld) [reply package.json]: Awesome; this is a great improvement.

File: node-tree-sitter.d.ts (hunk 3 above)
Comment (Xapphire13) [reply node-tree-sitter.d.ts]: In the code I only see seek taking in a single number argument, should it just be `seek(index: number): void;`?

Comment (maxbrunsfeld) [reply node-tree-sitter.d.ts]: Oh yeah, that's right. The `position` argument was something that I planned to add but never did.

**Comment by Xapphire13** [comment]
Would you mind completing the PR when you have a moment? I don't have access to the merge button =]

**Pull request merged**